mentat/include_files.py

```diff
@@ -1,38 +1,17 @@
-import glob
-import logging
+import fnmatch
 import os
-from collections import defaultdict
+import re
 from pathlib import Path
-from typing import Any, Dict
+from typing import Any, Iterable, List, Set
 
-from mentat.code_feature import CodeFeature, parse_intervals
-from mentat.git_handler import get_non_gitignored_files
+from mentat.code_feature import CodeFeature
+from mentat.errors import PathValidationError
+from mentat.git_handler import get_git_root_for_path, get_non_gitignored_files
+from mentat.interval import parse_intervals
 from mentat.session_context import SESSION_CONTEXT
 
 
-def expand_paths(paths: list[Path]) -> tuple[list[Path], list[str]]:
-    """Expand paths/globs into a list of absolute paths."""
-    globbed_paths = set[str]()
-    invalid_paths = set[str]()
-    for path in paths:
-        new_paths = glob.glob(pathname=str(path), recursive=True)
-        if new_paths:
-            globbed_paths.update(new_paths)
-        else:
-            split = str(path).rsplit(":", 1)
-            p = Path(split[0])
-            if len(split) > 1:
-                intervals = parse_intervals(split[1])
-            else:
-                intervals = None
-            if Path(p).exists() and intervals:
-                for interval in intervals:
-                    globbed_paths.add(f"{p}:{interval.start}-{interval.end}")
-            else:
-                invalid_paths.add(str(path))
-    return [Path(path).resolve() for path in globbed_paths], list(invalid_paths)
-
-
+# TODO: replace this with something that doesn't load the file into memory
```

Comment:
I don't think there's any other way to definitively check if a file is utf-8 encoded or not

Reply:
You can read it in chunks and validate each chunk. Out-of-scope for this PR anways